indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java

```diff
@@ -708,11 +844,77 @@ public void testRealtimeIndexTask() throws Exception{
     DataSegment segment = mdc.getPublished().iterator().next();
     Assert.assertEquals("test_ds", segment.getDataSource());
     Assert.assertEquals(ImmutableList.of("dim1", "dim2"), segment.getDimensions());
-    Assert.assertEquals(new Interval(now.toString("YYYY-MM-dd")+"/"+now.plusDays(1).toString("YYYY-MM-dd")), segment.getInterval());
+    Assert.assertEquals(
+        new Interval(now.toString("YYYY-MM-dd") + "/" + now.plusDays(1).toString("YYYY-MM-dd")),
+        segment.getInterval()
+    );
     Assert.assertEquals(ImmutableList.of("count"), segment.getMetrics());
     EasyMock.verify(monitorScheduler, serverView, queryRunnerFactoryConglomerate);
   }
 
+  @Test
+  public void testResumeTasks() throws Exception
+  {
+    final Task indexTask = new IndexTask(
+        null,
+        new IndexTask.IndexIngestionSpec(
+            new DataSchema(
+                "foo",
+                null,
+                new AggregatorFactory[]{new DoubleSumAggregatorFactory("met", "met")},
+                new UniformGranularitySpec(
+                    Granularity.DAY,
+                    null,
+                    ImmutableList.of(new Interval("2010-01-01/P2D"))
+                )
+            ),
+            new IndexTask.IndexIOConfig(new MockFirehoseFactory(false)),
+            new IndexTask.IndexTuningConfig(10000, 10, -1, indexSpec)
+        ),
+        TestUtils.MAPPER
+    );
+
+    final long startTime = System.currentTimeMillis();
+
+    // manually insert the task into TaskStorage, waiting for TaskQueue to sync from storage
+    ts.insert(indexTask, TaskStatus.running(indexTask.getId()));
+
+    while (tsqa.getStatus(indexTask.getId()).get().isRunnable()) {
```

Comment:
I don't think it's doable here though, unless there is a way to attach a callback for task completion. This is actually the same way of waiting for task completion as in ```runTask()```, I think ```Thread.sleep(100)``` here aims to relieve busy wait a bit.